starwars_api/models.py

```diff
@@ -55,27 +63,64 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.next_page = None
+        self.previous_page = None
+        self.results = []
+        self.current_record = 0
+        self.current_page = 0
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
-
+        while True:
+            if not self.results:
```

Comment:
`self.results` contains an empty list to begin with. `if not self.results` in this case is analogous to `if self.results == []`. Although if `self.results == None` will also fit the case, we never allow results to be `None` so we know that will not happen.